Sources/NIOCore/ByteBuffer-views.swift

```diff
@@ -158,6 +158,39 @@ extension ByteBufferView: RangeReplaceableCollection {
         self = ByteBufferView(ByteBuffer())
     }
 
+    /// Reserves enough space in the underlying `ByteBuffer` such that this view can
+    /// store the specified number of bytes without reallocation.
+    ///
+    /// See the documentation for ``ByteBuffer.reserveCapacity(_:)`` for more details.
+    @inlinable
+    public mutating func reserveCapacity(_ minimumCapacity: Int) {
+        let additionalCapacity = minimumCapacity - self.count
+        if additionalCapacity > 0 {
+            self._buffer.reserveCapacity(self._buffer.capacity + additionalCapacity)
+        }
+    }
+
+    /// Writes a single byte to the underlying `ByteBuffer`.
+    @inlinable
+    public mutating func append(_ byte: UInt8) {
+        // ``CollectionOfOne`` has no witness for
+        // ``Sequence.withContiguousStorageIfAvailable(_:)``. so we do this instead:
+        Swift.withUnsafeBytes(of: byte)
```

Comment:
We don't need unsafe code here, we can just use `writeInteger(byte)`.

Reply:
replaced with `setInteger` per discussion below